neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -212,7 +212,9 @@ def prepare_chat_context(self, chat_message_history: List[Dict[str, Any]]) -> Di
                 the conversation such that it could be taken up on a different
                 server instance
         """
-        processor: MessageProcessor = ChatHistoryMessageProcessor()
+        # OK if this is None
+        max_message_history: int = self.front_man.get_agent_tool_spec().get("max_message_history")
+        processor: MessageProcessor = ChatHistoryMessageProcessor(max_message_history)
```

Comment:
Get the max_message_history from the front-man.

Reply:
You should document it somewhere. No way I remember if someone asks.

Reply:
I think there is a test case to be had (see beatles example above) when the infrastructure comes.
At some point it will be worth putting together a pretty extensive hocon document. Would be good to see what we have going in demos before starting on that.